bitcoind-rpc/src/main/scala/org/bitcoins/rpc/client/common/Client.scala

```diff
@@ -330,18 +330,22 @@ trait Client
   /** Cached http client to send requests to bitcoind with */
   private lazy val httpClient: HttpExt = Http(system)
 
-  private lazy val httpConnectionPoolSettings: ConnectionPoolSettings =
+  private lazy val httpConnectionPoolSettings: ConnectionPoolSettings = {
     instance.proxyParams match {
       case Some(proxyParams) =>
-        val socks5ClientTransport = new Socks5ClientTransport(proxyParams)
+        if (instance.rpcUri.getHost.contains(".onion")) {
```

Comment:
I think we should connect over tor for all kind of addresses if `proxy.enabled=true`
Otherwise it will leak the user's data when running over clear net.

As far as I understand, this check is needed to avoid connection errors when BTC RPC backend is enabled and running on `localhost`, but tor cannot proxy connections to `localhost` and fails. 

So it looks like it's more appropriate just to check if  rpcUri is pointing to `localhost` (`127.0.0.1` or `::1`) and don't set up a proxy in this case.